samples/matter/lock/README.rst

```diff
@@ -329,6 +329,55 @@ Upgrading the device firmware
 
 To upgrade the device firmware, complete the steps listed for the selected method in the :doc:`matter:nrfconnect_examples_software_update` tutorial of the Matter documentation.
 
+Switching between Thread and Wi-Fi
+==================================
+
+The ``thread_wifi_switched`` build type allows you to build the door lock application that supports dynamic switching between Matter over Thread and Matter over Wi-Fi.
+This feature requires that both variants of the application be built and placed in appropriate partitions of the external flash.
+After that, when a user triggers the switch, the device is rebooted into the MCUboot bootloader which swaps the current variant.
+
+To test switching between Thread and Wi-Fi, complete the following steps:
+
+#. Build the door lock application for Matter over Thread:
+
+   .. code-block:: console
+
+      west build -b nrf5340dk_nrf5340_cpuapp -- -DCONF_FILE=prj_thread_wifi_switched.conf -DSHIELD=nrf7002_ek -DCONFIG_CHIP_WIFI=n
+
+#. Program the application to the kit using the following command:
+
+   .. code-block:: console
+
+      west flash --erase
+
+#. Program the application to a partition of the external flash using the following command:
+
+   .. code-block:: console
+
+      west build -t flash-external
+
+#. Build the door lock application for Matter over Wi-Fi:
+
+   .. code-block:: console
+
+      west build -b nrf5340dk_nrf5340_cpuapp -p always -- -DCONF_FILE=prj_thread_wifi_switched.conf -DSHIELD=nrf7002_ek
+
+#. Program the application to another partition of the external flash using the following command:
+
+   .. code-block:: console
+
+      west build -t flash-external
+
+#. |connect_terminal_ANSI|
+#. At any time, press **Button 3** for more than ten seconds to trigger switching to the Matter over Wi-Fi variant of the application.
```

Comment:
```suggestion
#. Press **Button 3** for more than ten seconds to trigger switching to the Matter over Wi-Fi variant of the application.
```
Is there a different way to achieve this than pressing for "more than 10 seconds"? I feel that even just "more than 3 seconds" can be very confusing for users, and 10 seconds sounds like eternity...

Reply:
There's no. I guess Kamil has chosen 10 seconds as even when using 6 seconds to trigger the factory reset we got reports that someone had triggered that unintentionally... I'm happy to discuss our UI in the standups though.

Reply:
Yes 10 seconds was chosen by me, as a value big enough to not do that accidentally. Please note that 10 s is often selected value for such critical changes, e.g. in case of factory reset of Google and Apple hubs, so maybe the UX will not be that bad.